tests/integration/test_migrator.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+import os
+import pkg_resources
+import zlib
+
+import pytest
+from flask import current_app
+from redis import StrictRedis
+
+from invenio_db import db
+from invenio_pidstore.models import PersistentIdentifier
+
+from inspirehep.modules.migrator.models import InspireProdRecords
+from inspirehep.modules.migrator.tasks.records import continuous_migration
+from inspirehep.utils.record_getter import get_db_record
+
+
+def _delete_record(pid_type, pid_value):
+    get_db_record(pid_type, pid_value)._delete(force=True)
+
+    pid = PersistentIdentifier.get(pid_type, pid_value)
+    PersistentIdentifier.delete(pid)
+
+    object_uuid = pid.object_uuid
+    PersistentIdentifier.query.filter(
+        object_uuid == PersistentIdentifier.object_uuid).delete()
+
+    db.session.commit()
+
+
+def push_to_redis(record_file):
+    record = pkg_resources.resource_string(
+        __name__, os.path.join('fixtures', record_file))
+
+    redis_url = current_app.config.get('CACHE_REDIS_URL')
```

Comment:
This is the most notable change: since this method will be executed when an application context is available, we can use `current_app` instead of passing around the `app` fixture.